### PR TITLE
Lxc consistency2

### PR DESCRIPTION
**Pull request by kiorky**
depends on #24065
refs #23657 
blocks #23982

### CODE REVIEW

**Comment by kiorky** [comment]
this should wait that we sync with @basepi to merge

File: salt/modules/lxc.py [hunk 1]
```diff
@@ -800,6 +835,14 @@ def _network_conf(conf_tuples=None, **kwargs):
...
         for row in val:
             ret.append(salt.utils.odict.OrderedDict([(row, val[row])]))
+    # on old versions of lxc, still support the gateway auto mode
+    # if we didnt explicitly say no to
+    # (lxc.network.ipv4.gateway: auto)
+    if (
+        distutils.version.LooseVersion(version()) <= '1.0.7' and
+        True not in ['ipv4.gateway' in a[0] for a in ret]
+    ):
+        ret.append({'lxc.network.ipv4.gateway': 'auto'})
```
Comment (cheuschober) [inline salt/modules/lxc.py]: @kiorky, I encountered a KeyError here, the value of `ret` prior to falling into the comprehension is:

```python
[OrderedDict([('lxc.network.type', 'veth')]),
 OrderedDict([('lxc.network.name', 'eth0')]),
 OrderedDict([('lxc.network.flags', 'up')]),
 OrderedDict([('lxc.network.link', 'lxcbr0')]),
 OrderedDict([('lxc.network.hwaddr', 'AC:DE:48:0D:88:79')])]
```
I think the comprehension be `True not in ['ipv4.gateway' in a for a in ret]`

Comment (kiorky) [inline salt/modules/lxc.py]: yep.

Comment (kiorky) [inline salt/modules/lxc.py]: uhm, wait no. im verifying.

Comment (kiorky) [inline salt/modules/lxc.py]: @cheuschober pushed.

File: doc/topics/tutorials/lxc.rst [hunk 2]
```diff
@@ -227,6 +237,41 @@ container-by-container basis, for instance using the ``nic_opts`` argument to
...
 
 
+Old lxc support (<1.0.7)
+---------------------------
+
+With saltstack **2015.5.2** and above, normally the setting is autoselected, but
+before, you ll need to teach your network profile to set
+**lxc.network.ipv4.gateway** to **auto** when using a classic ipv4 configuration.
+
+Thus you ll need
```
Comment (basepi) [inline doc/topics/tutorials/lxc.rst]: s/you ll/you'll/

File: salt/modules/lxc.py [hunk 3]
```diff
@@ -583,6 +619,14 @@ def get_network_profile(name=None):
             type: veth
             flags: up
 
+    To disable entirely net:
```
Comment (basepi) [inline salt/modules/lxc.py]: s/To disable entirely net/To disable networking entirely/

**Comment by basepi** [comment]
I am going to just merge this and make the two changes I noted above separately.

**Pull request merged**